demo_apps/messenger_llama2.md

```diff
@@ -0,0 +1,192 @@
+  express = require("express"),
+  body_parser = require("body-parser"),
+  axios = require("axios").default,
+  app = express().use(body_parser.json()); // creates express http server
+
+// Sets server port and logs message on success
+app.listen(process.env.PORT || 1337, () => console.log("webhook is listening"));
+
+// Accepts POST requests at /webhook endpoint
+app.post("/webhook", (req, res) => {
+  // Parse the request body from the POST
+  let body = req.body;
+  
+  let sender = req.body["entry"][0]["messaging"][0]['sender']['id']
+  let recipient = req.body["entry"][0]["messaging"][0]['recipient']['id']
+  let message = req.body["entry"][0]["messaging"][0]['message']['text']
+
+  // Check if this is an event from a page subscription
+  if (body.object === "page") {
+    // Returns a '200 OK' response to all requests
+    res.status(200).send("EVENT_RECEIVED");
+    
+    let url = "http://<web server public IP>:5000/msgrcvd_page?sender=" + sender + "&recipient=" + recipient + "&message=" + encodeURIComponent(message)
+    console.log(url)
+  
+    axios.get(url)
+      .then(response => {
+        // Handle the response data
+        console.log(response.data);
+      })
+      .catch(error => {
+        // Handle errors
+        console.error('Axios error:', error);
+      });    
+    } else {
+      // Return a '404 Not Found' if event is not from a page subscription
+      res.sendStatus(404);
+    }
+  });
+
+// Accepts GET requests at the /webhook endpoint. You need this URL to setup webhook initially.
+// info on verification request payload: https://developers.facebook.com/docs/graph-api/webhooks/getting-started#verification-requests 
+app.get("/webhook", (req, res) => {
+  /**
+   * UPDATE YOUR VERIFY TOKEN
+   *This will be the Verify Token value when you set up webhook
+  **/
+  const verify_token = process.env.VERIFY_TOKEN;
+
+  // Parse params from the webhook verification request
+  let mode = req.query["hub.mode"];
+  let token = req.query["hub.verify_token"];
+  let challenge = req.query["hub.challenge"];
+
+  // Check if a token and mode were sent
+  if (mode && token) {
+    // Check the mode and token sent are correct
+    if (mode === "subscribe" && token === verify_token) {
+      // Respond with 200 OK and challenge token from the request
+      console.log("WEBHOOK_VERIFIED: " + token);
+      res.status(200).send(challenge);
+    } else {
+      // Responds with '403 Forbidden' if verify tokens do not match
+      res.sendStatus(403);
+    }
+  }
+});
+
+```
+Remember to change <web server public IP>, which needs to be publicly visible, to the IP of the server where your Llama 2 enabled web app in the previous section runs.
+
+## Running the Chatbot
+
+On your web server, run the following command on a Terminal:
+
+```
+gunicorn -b 0.0.0.0:5000 llama_messenger:app
+```
+
+If you use Amazon EC2 as your web server, make sure you have port 5000 added to your EC2 instance's security group's inbound rules. 
```

Comment:
Can the scope of access be narrowed to improve security? ip range or similar? Is there a tutorial we can reference for this?

Reply:
i'm not sure but added the Gunicorn doc.